src/openms/source/QC/DBSuitability.cpp

```diff
@@ -597,6 +616,78 @@ namespace OpenMS
     return -(db_slope) / (deNovo_slope);
   }
 
+  UInt DBSuitability::numberOfUniqueProteins_(const std::vector<PeptideIdentification>& peps, UInt number_of_hits) const
+  {
+    set<String> proteins;
+
+    for (const auto& pep : peps)
+    {
+      const vector<PeptideHit>& hits = pep.getHits();
+      for (UInt i = 0; i < number_of_hits; ++i)
+      {
+        const PeptideHit& hit = hits[i];
+
+        // skip if the hit is a decoy hit
+        if (!hit.metaValueExists("target_decoy"))
+        {
+          throw(Exception::MissingInformation(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "No target/decoy information found! Make sure 'PeptideIndexer' is run beforehand."));
+        }
+        if (hit.getMetaValue("target_decoy") == "decoy") continue;
+
+        // insert protein accessions
+        const set<String> accessions = hit.extractProteinAccessionsSet();
+        for (const String& acc : accessions)
+        {
+          String acc_copy(acc);
+          if (acc.find(Constants::UserParam::CONCAT_PEPTIDE) != String::npos) // skip novo accessions
+          {
+            continue;
+          }
+          if (boost::regex_search(acc_copy.toLower(), decoy_pattern_)) // skip decoy accessions
+          {
+            continue;
+          }
+
+          proteins.insert(acc); // insert the rest
+        }
+      }
+    }
+
+    return proteins.size();
+  }
+
+  DBSuitability::SuitabilityData DBSuitability::getEntryWithMedianNovoHits_(const vector<SuitabilityData>& data) const
+  {
+    if (data.size() == 0)
+    {
+      throw(Exception::IllegalArgument(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "No suitability data given!"));
+    }
+
+    vector<Size> novo_data;
+    map<Size, Size> novo_hits_to_data;
+    for (int i = 0; i < data.size(); ++i)
+    {
+      Size num_top_novo = data[i].num_top_novo;
+      novo_data.push_back(num_top_novo);
+      novo_hits_to_data[num_top_novo] = i;
+    }
+
+    UInt curr_index = 0;
+    double best_diff = DBL_MAX;
+    double avg_novo_hits = Math::mean(novo_data.begin(), novo_data.end());
```

Comment:
median is more robust?!
If the number of items is even, it's a bit weird, because you must decide which of both 'median' items to take (they are equally far away from the computed median). Thus you either always underestimate, or always overestimate...
The only solution is to sample an odd number of times. Then the median is well defined.

Reply:
I use the mean to calculate the median by hand. Because I want the value which is closest to the mean, right? `Math::median` just gives me the value which is located in the middle of the sorted list, which isn't always the same. Am I wrong?

Reply:
somewhat...
the median can be miles away from the mean... so better not trust the mean in any way.